src/Job/Message.php

```diff
@@ -120,10 +120,10 @@ public function getCallable()
     public function getArgument($key = null, $default = null)
     {
         if ($key === null) {
-            return $this->parsedBody['args'][0];
+            return $this->parsedBody['args'];
         }
 
-        return Hash::get($this->parsedBody['args'][0], $key, $default);
+        return Hash::get($this->parsedBody['args'], $key, $default);
```

Comment:
If we move forward with this, we should be able to handle old and new messages. This is important as during a deploy you'll have application instances emitting the old message format, and messages being consumed on a new worker process. By handling the old format and the new format we can provide users with a zero-downtime upgrade.

I would likely recommend using a different key than `args` for the new format. Because there will be argument lists that contain positional arguments a simple count check won't work. This will also necessitate emitting the `args` key as well because of cross talk during deploys.

Reply:
I see what you are saying. I'm not quite sure I agree that we need to provide a zero-downtime option for this since the package is still in beta, but I will take a stab at writing some code to support both.

Reply:
I went with the key `data`. I added a check to see if `args` exists and will use that. If `args` does not exist in the payload `data` is used. I updated all variables to be `$data` when dealing with this to denote the change.

Reply:
> I see what you are saying. I'm not quite sure I agree that we need to provide a zero-downtime option for this since the package is still in beta, but I will take a stab at writing some code to support both.

Even though we're still in beta its good to think about the upgrade process, as if we can make it smooth and painless for folks who update frequently then they will continue to update.